version: '3.7'

services:

  web:
    build: .
    depends_on:
      db:
        condition: service_healthy
    command: python manage.py runserver 0.0.0.0:8000
    env_file:
      - .env
    environment:
      POSTGRES_USER: $POSTGRES_USER
      POSTGRES_PASSWORD: $POSTGRES_PASSWORD
      POSTGRES_DB: $POSTGRES_DB
    restart: always
    networks:
      - web_net
    volumes:
      - .:/code/
    ports:
      - 8000:8000
    container_name: python_django_team-web  

  db:
    image: postgres:13
    environment:
      POSTGRES_USER: $POSTGRES_USER
      POSTGRES_PASSWORD: $POSTGRES_PASSWORD
      POSTGRES_DB: $POSTGRES_DB
    restart: always
    networks:
      - web_net
    ports:
      - $POSTGRES_PORT:5432
    volumes:
      - postgres_data:/var/lib/postgresql/data
    container_name: python_django_team-postgres_db  
    healthcheck:
      test: [ "CMD", "pg_isready", "-q", "-d", "${POSTGRES_DB}", "-U", "${POSTGRES_USER}" ]
      timeout: 45s
      interval: 10s
      retries: 10

  redis_db:
    image: redis:alpine
    restart: always
    volumes:
      - redis_data:/data
    networks:
      - web_net
    container_name: python_django_team-redis_db  

  celeryw:
    restart: always
    build:
      context: .
    command: celery -A payment worker -l info
    volumes:
      - .:/code/
    depends_on:
      - db
      - redis_db
      - web
    networks:
      - web_net  
    container_name: python_django_team-celery_worker

  celeryb:
    restart: always
    build:
      context: .
    command: celery -A payment beat -l info -S django
    volumes:
      - .:/code/
    depends_on:
      - db
      - redis_db
      - web
    networks:
      - web_net  
    container_name: python_django_team-celery_beat

volumes:
  postgres_data:
  redis_data:

networks:
  web_net:
    name: web_net
    driver: bridge
